ui/src/reducers/entities.js

```diff
@@ -20,6 +21,8 @@ export default createReducer({
 
   [queryEntities.COMPLETE]: (state, { result }) => resultObjects(state, result),
 
+  [undeleteEntity.COMPLETE]: (state, { id, data }) => objectLoadComplete(state, id, data),
+
   [deleteEntity.COMPLETE]: (state, { id }) => objectDelete(state, id),
 
 }, initialState);
```

Comment:
I think we may need to also hook this into the `mutation` reducer to flush front-end cache....